core/templates/services/navigation.service.ts

```diff
@@ -13,15 +13,34 @@
 // limitations under the License.
 
 /**
- * @fileoverview Factory for navigating the top navigation bar with
+ * @fileoverview Service for navigating the top navigation bar with
  * tab and shift-tab.
  */
-angular.module('oppia').factory('NavigationService', [function() {
-  var navigation = {
-    activeMenuName: '',
-    ACTION_OPEN: 'open',
-    ACTION_CLOSE: 'close',
-    KEYBOARD_EVENT_TO_KEY_CODES: {
+
+
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+interface KeyboardEvent {
+  keyCode?: number,
+  shiftKey?: boolean,
+  currentTarget?
+}
+
+interface EventToCodes {
+  enter?,
+  tab?,
+  shiftTab?,
+}
```

Comment:
export these interfaces and reuse them in the `top-navigation-bar.component.ts` instead of redefining the same there.

Reply:
I get this when logging `eventsTobeHandled` to the console.
So, string type here.

Reply:
you don't have to create an interface for `KeyboardEvent`. It is already present in typescript `node_modules/typescript/lib/lib.dom.d.ts` to be exact.

Reply:
done